src/MergeContext.js

```diff
@@ -1295,6 +1314,9 @@ class PullRequest {
             else
                 this._removePositiveStagingLabels();
 
+            // some of these labels may be set already in the exception
+            this._addPrStateLabels();
```

Comment:
I think we should delete/inline the used-once _addPrStateLabels()  method here. We are kind of working in a dangerous spot here -- handling any exception -- and I am worried that we will forget about it and start adding advanced code into that method if it will be called from somewhere else as well. Inlining will also clarify our logic, at least for now.